### PR TITLE
added the new Variables.md file

### LINKED ISSUE #1243: [Concept Entry] Go: Variables
### Reason/inspiration (optional)

We would like a new concept entry for [Go](https://www.codecademy.com/resources/docs/go) about variables.

The entry should go in a new file in docs/content/go/concepts/variables/variables.md

The entry should

- Explain what variables are.
- Explain the syntax of declaring a variable.

### Entry Type (select all that apply)

New Concept Entry for Existing Topic

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [X] Labels added
- [x] Issue is assigned

### PR DESCRIPTION
**Pull request by Abdelrahmanabusaa**
### Description

I have added a new Variables folder in docs/content/go/concepts path that contains a markdown file with a new concept to describe Variables in Go. 

Closes #1243 

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/go/concepts/Variables/Variables.md
Comment (SSwiniarski) [inline content/go/concepts/Variables/Variables.md:4]:
```suggestion
Description: 'Variables are used to store numbers, strings, true/false boolean states, or objects based on their type when declaring them.'
```

File: content/go/concepts/Variables/Variables.md
Comment (SSwiniarski) [inline content/go/concepts/Variables/Variables.md:24]:
```suggestion
Unlike constants, when declaring a variable, its value can be reassigned later on but with restriction that the new value's data type is the same as the data type when initialized.
```

File: content/go/concepts/Variables/Variables.md
Comment (SSwiniarski) [inline content/go/concepts/Variables/Variables.md:26]:
```suggestion
### Pseudocode Example
```

File: content/go/concepts/Variables/Variables.md
Comment (SSwiniarski) [inline content/go/concepts/Variables/Variables.md:44]:
```suggestion
## Syntax
```

File: content/go/concepts/Variables/Variables.md
Comment (SSwiniarski) [inline content/go/concepts/Variables/Variables.md:50]:
```suggestion
Begin with the `var` keyword followed by the name of the variable, and at the end the variable type. Each of these are separated by spaces.
```

File: content/go/concepts/Variables/Variables.md
Comment (SSwiniarski) [inline content/go/concepts/Variables/Variables.md:52]:
```suggestion
Variable names should:
```

File: content/go/concepts/Variables/Variables.md
Comment (SSwiniarski) [inline content/go/concepts/Variables/Variables.md:54]:
```suggestion
- Begin with a letter. 
```

File: content/go/concepts/Variables/Variables.md
Comment (SSwiniarski) [inline content/go/concepts/Variables/Variables.md:55]:
```suggestion
- Must not contain special characters, only letters and digits.
```

File: content/go/concepts/Variables/Variables.md
Comment (SSwiniarski) [inline content/go/concepts/Variables/Variables.md:57]:
```suggestion
- Names are case sensitive.
```

File: content/go/concepts/Variables/Variables.md
Comment (SSwiniarski) [inline content/go/concepts/Variables/Variables.md:59]:
```suggestion
Some of the basic types are:
```

File: content/go/concepts/Variables/Variables.md
Comment (SSwiniarski) [inline content/go/concepts/Variables/Variables.md:79]:
```suggestion
```

File: content/go/concepts/Variables/Variables.md
Comment (SSwiniarski) [inline content/go/concepts/Variables/Variables.md:69]:
```suggestion
### Initializing After Declaring
```

File: content/go/concepts/Variables/Variables.md
Comment (SSwiniarski) [inline content/go/concepts/Variables/Variables.md:67]:
```suggestion
To declare a variable you have multiple ways with syntax similar to other programming languages.
```

File: content/go/concepts/Variables/Variables.md
Comment (SSwiniarski) [inline content/go/concepts/Variables/Variables.md:56]:
```suggestion
- Must not be a Go keyword. 
```

File: content/go/concepts/Variables/Variables.md
Comment (SSwiniarski) [inline content/go/concepts/Variables/Variables.md:22]:
```suggestion
## Difference From Constants 
```

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
@Abdelrahmanabusaa I did a first review and left a number of comments.

File: content/go/concepts/Variables/Variables.md
Comment (HishamT) [inline content/go/concepts/Variables/Variables.md:20]:
```suggestion
Variables are used to store data such as numbers, strings, or Booleans in memory for later use.
```

File: content/go/concepts/Variables/Variables.md
Comment (HishamT) [inline content/go/concepts/Variables/Variables.md:31]:
```suggestion
declare an int variable
```

File: content/go/concepts/Variables/Variables.md
Comment (HishamT) [inline content/go/concepts/Variables/Variables.md:39]:
```suggestion
declare an int variable
```

File: content/go/concepts/Variables/Variables.md
Comment (HishamT) [inline content/go/concepts/Variables/Variables.md:41]: Is this supposed to be "Codecaedmy" or did you mean "Codecademy"?

File: content/go/concepts/Variables/Variables.md
Comment (HishamT) [inline content/go/concepts/Variables/Variables.md:61]: This line and the following two should be in backticks as they are Go keywords.

Example: `int`.

File: content/go/concepts/Variables/Variables.md
Comment (HishamT) [reply content/go/concepts/Variables/Variables.md:67]: @SSwiniarski, I believe there should also be a comma after "variable".

"To declare a variable, you have multiple ways..."

File: content/go/concepts/Variables/Variables.md
Comment (HishamT) [reply content/go/concepts/Variables/Variables.md:108]: @SSwiniarski, should this be "This way skips all *of* the syntax..."?

File: content/go/concepts/Variables/Variables.md
Comment (HishamT) [reply content/go/concepts/Variables/Variables.md:114]: @SSwiniarski, shouldn't this be "Initializing Multiple Variables"?

**Review by HishamT — CHANGES_REQUESTED** [review]
Hey @Abdelrahmanabusaa, I have finished the second review of your entry and left you some comments. :grin:

File: content/go/concepts/Variables/Variables.md
Comment (SSwiniarski) [inline content/go/concepts/Variables/Variables.md:75]:
```suggestion
```go
```

File: content/go/concepts/Variables/Variables.md
Comment (SSwiniarski) [inline content/go/concepts/Variables/Variables.md:88]:
```suggestion
```go
```

File: content/go/concepts/Variables/Variables.md
Comment (SSwiniarski) [inline content/go/concepts/Variables/Variables.md:99]:
```suggestion
```go
```

File: content/go/concepts/Variables/Variables.md
Comment (SSwiniarski) [inline content/go/concepts/Variables/Variables.md:108]:
```suggestion
```go
```

File: content/go/concepts/Variables/Variables.md
Comment (SSwiniarski) [inline content/go/concepts/Variables/Variables.md:116]:
```suggestion
```go
```

**Review by SSwiniarski — APPROVED** [review]
@Abdelrahmanabusaa, I added a few more comments, otherwise this is ready for a second review.

File: content/go/concepts/Variables/Variables.md
Comment (HishamT) [inline content/go/concepts/Variables/Variables.md:24]:
```suggestion
Unlike constants, when declaring a variable, its value can be reassigned later on but with the restriction that the new value's data type is the same as the data type when initialized.
```

File: content/go/concepts/Variables/Variables.md
Comment (HishamT) [inline content/go/concepts/Variables/Variables.md:67]:
```suggestion
There are multiple ways to declare a variable. The syntax is similar to other programming languages.
```

File: content/go/concepts/Variables/Variables.md
Comment (HishamT) [inline content/go/concepts/Variables/Variables.md:71]:
```suggestion
This way is used if a group of variables is needed but will be assigned values later in the program.
```

File: content/go/concepts/Variables/Variables.md
Comment (HishamT) [inline content/go/concepts/Variables/Variables.md:106]:
```suggestion
This way skips almost all of the syntax and only needs the name of the variable followed by `:=`.
```

File: content/go/concepts/Variables/Variables.md
Comment (HishamT) [inline content/go/concepts/Variables/Variables.md:114]:
```suggestion
This way is used to declare multiple variables on the same line. Variables are separated with a comma (',').
```

**Review by HishamT — COMMENTED** [review]
I left some more comments. Should be ready after that.

File: content/go/concepts/Variables/Variables.md
Comment (Dusch4593) [inline content/go/concepts/Variables/Variables.md:20]:
```suggestion

**Variables** are used to store values with specific [data types](https://www.codecademy.com/resources/docs/go/data-types) (e.g., numbers, strings, or booleans) in memory for later use.
```

File: content/go/concepts/Variables/Variables.md
Comment (Dusch4593) [inline content/go/concepts/Variables/Variables.md:24]:
```suggestion
## Difference From Constants 

Unlike constants, the value of variables can be reassigned after declaration, but its data type must remain the same as it was when first initialized. For example, an `int` variable can start as 2 and be reassigned as 0 (another `int`), but it can't be reassigned as "Codecademy" (a `string`).
```

File: content/go/concepts/Variables/Variables.md
Comment (Dusch4593) [inline content/go/concepts/Variables/Variables.md:42]:
```suggestion
```

File: content/go/concepts/Variables/Variables.md
Comment (Dusch4593) [inline content/go/concepts/Variables/Variables.md:46]:
```suggestion
```pseudo
```

File: content/go/concepts/Variables/Variables.md
Comment (Dusch4593) [inline content/go/concepts/Variables/Variables.md:47]:
```suggestion
var name type
```

File: content/go/concepts/Variables/Variables.md
Comment (Dusch4593) [inline content/go/concepts/Variables/Variables.md:50]:
```suggestion
Variable statements begin with the `var` keyword, followed by its `name`, and then the variable `type`.
```

File: content/go/concepts/Variables/Variables.md
Comment (Dusch4593) [inline content/go/concepts/Variables/Variables.md:57]:
```suggestion
Variable names begin with a letter and are case-sensitive. They must not contain special characters, only letters and digits. They must also not be a reserved Go keyword such as `if` or `for`.
```

File: content/go/concepts/Variables/Variables.md
Comment (Dusch4593) [inline content/go/concepts/Variables/Variables.md:4]:
```suggestion
Description: 'Variables are used to store values of specific data types in memory for later use.'
```

File: content/go/concepts/Variables/Variables.md
Comment (Dusch4593) [inline content/go/concepts/Variables/Variables.md:64]:
```suggestion
```

If we link to the existing entry on Go data types, this part wouldn't be necessary.

File: content/go/concepts/Variables/Variables.md
Comment (Dusch4593) [inline content/go/concepts/Variables/Variables.md:124]:
```suggestion
```

Too many trailing spaces.

File: content/go/concepts/Variables/Variables.md
Comment (Dusch4593) [inline content/go/concepts/Variables/Variables.md:114]:
```suggestion
This way is used to declare multiple variables on the same line. Variables are separated with a comma (`,`):
```

File: content/go/concepts/Variables/Variables.md
Comment (Dusch4593) [inline content/go/concepts/Variables/Variables.md:106]:
```suggestion
This way skips almost all of the syntax and only needs the name of the variable followed by `:=`:
```

File: content/go/concepts/Variables/Variables.md
Comment (Dusch4593) [inline content/go/concepts/Variables/Variables.md:98]:
```suggestion
This way lets the compiler assign the type for the input without declaring it:

```

File: content/go/concepts/Variables/Variables.md
Comment (Dusch4593) [inline content/go/concepts/Variables/Variables.md:87]:
```suggestion
### Initializing With Declaring

This way is used to assign values directly when defining a variable:
```

File: content/go/concepts/Variables/Variables.md
Comment (Dusch4593) [inline content/go/concepts/Variables/Variables.md:74]:
```suggestion
### Initializing After Declaring

This way is used if a group of variables is needed but will be assigned values later in the program:
```

**Review by Dusch4593 — CHANGES_REQUESTED** [review]
@Abdelrahmanabusaa Finished my review and wanted to offer some comments/suggestions. Also, could we rename the folder and file from `Variables` and `Variables.md` to `variables` and `variables.md`, respectively?

**Review by Dusch4593 — APPROVED** [review]
@Abdelrahmanabusaa LGTM! 👍🏻

I made a few changes in f0ea278de1b40f5671977e39cda0f47d28118df2, including the following:

- Light edit on line # 21.
- Added empty line above a few \```go blocks.
- Removed a few trailing spaces.

**Comment by Dusch4593** [comment]
Congrats and thank you for contributing to Codecademy Docs, @Abdelrahmanabusaa! 🎉 🎊 🥳

The entry should now be live: https://www.codecademy.com/resources/docs/go/variables

Currently, your Codecademy profile seems to be anonymously listed on the entry page. To make it visible, you'll need to add your GitHub username to your Codecademy profile, log out of Codecademy, and log back in through your GitHub account:

Screen Shot 2022-11-22 at 1 35 54 PM

Looking forward to your next Docs contribution! 😄

**Pull request merged**
